src/github.com/t3chguy/riot-static/matrix-client/room.go

```diff
@@ -1,328 +0,0 @@
-// Copyright 2017 Michael Telatynski <7t3chguy@gmail.com>
-//
-// Licensed under the Apache License, Version 2.0 (the "License");
-// you may not use this file except in compliance with the License.
-// You may obtain a copy of the License at
-//
-//     http://www.apache.org/licenses/LICENSE-2.0
-//
-// Unless required by applicable law or agreed to in writing, software
-// distributed under the License is distributed on an "AS IS" BASIS,
-// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-// See the License for the specific language governing permissions and
-// limitations under the License.
-
-package matrix_client
-
-import (
-	"fmt"
-	"github.com/matrix-org/gomatrix"
-	"github.com/t3chguy/riot-static/utils"
-	"sort"
-	"strings"
-	"sync"
-	"time"
-)
-
-type RoomEventErrorEnum int
-
-const (
-	RoomEventsCouldNotFindEvent RoomEventErrorEnum = iota
-	RoomEventsUnknownError
-	RoomEventsFine
-)
-
-type Room struct {
-	client *Client // each room has a client that is responsible for its state being up to date
-
-	// Active lock is called for the duration of a Read/Write on any field of the room.
-	// This lock may be Locked/Unlocked several times in a request, so may be hit by inconsistent data,
-	// for this reason requests which may move the anchor point of our eventList/latestRoomState Lock requestLock (TOO?)
-	activeLock sync.RWMutex
-
-	// Request lock is called for the duration of a request on the room object
-	// To prevent the CRON forward pagination from changing the 0 point of a timeline (and mangling latestRoomState)
-	// Write Lock when modifying the start of the eventList, Read Lock when wanting to prevent such modification.
-	requestLock sync.RWMutex
-
-	ID string // IMMUTABLE
```

Comment:
Strings in Go are immutable, yes. This is not a very useful comment. It's akin to saying `ID uint // NOT NEGATIVE` and is just distracting.